File: itinerary_server/src/main/java/com/dametto/itinerary_server/jpa/Post.java
package com.dametto.itinerary_server.jpa;

import java.util.Date;
import java.util.List;

public class Post {
    private Integer id;

    private ShallowUser author;

    private String body;

    private Date createdAt;

    private List<Comment> comments;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public ShallowUser getAuthor() {
        return author;
    }

    public void setAuthor(ShallowUser author) {
        this.author = author;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public void setComments(List<Comment> comments) {
        this.comments = comments;
    }
}

File: itinerary-fe/src/app/models/post.ts
import * as moment from "moment";
import { Comment } from "./comment";
import { User } from "./user";

export class Post {
    public id: number;
    public author: User;
    public body: string;
    public createdAt: moment.Moment;
    public comments: Comment[];

    constructor(id: number, author: User, body: string, createdAt: moment.Moment, comments: Comment[]) {
        this.id = id;
        this.author = author;
        this.body = body;
        this.createdAt = createdAt;
        this.comments = comments;
    }
}